lib/Command/ExpirePassword.php

```diff
@@ -106,54 +130,96 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+		$uids = $input->getOption('uid');
 
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
-
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
+		$expireDate = new \DateTime();
+		$oldDate = new \DateTime();
+		$timeZone = new \DateTimeZone('UTC');
+		$inputExpireDate = $input->getArgument('expiredate');
+
+		if ($allUsers !== false) {
+			$this->userManager->callForAllUsers(function (IUser $user) use ($expireDate, $oldDate, $timeZone, $inputExpireDate, $output) {
+				$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+				$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+			});
+		} elseif (\count($groups) >= 1) {
```

Comment:
It might be useful to expire passwords for groups and users at the same time. In addition, the help doesn't say anything that suggest that this can't be done